safe/impact_functions/base.py

```diff
@@ -208,6 +234,26 @@ def actual_extent_crs(self):
         return self._actual_extent_crs
 
     @property
+    def map_canvas(self):
```

Comment:
Is it ok to put map_canvas in impact function base? Please correct me, but in my mind, map_canvas is more related to the QGIS not the impact function it self.